src/plugins/fileview/fileview.tsx

```diff
@@ -0,0 +1,211 @@
+import React from "react";
+import * as T from "../../types/types";
+import { debounce } from "throttle-debounce";
+import * as mobx from "mobx";
+import * as mobxReact from "mobx-react";
+import { parse } from "path";
+import { For } from "tsx-control-statements/components";
+import { PacketDataBuffer } from "../core/ptydata";
+import { boundMethod } from "autobind-decorator";
+import { GlobalModel } from "../../models";
+import * as path from "path";
+
+import "./fileview.less";
+
+type OV<V> = mobx.IObservableValue<V>;
+
+class FileViewRendererModel {
+    context: T.RendererContext;
+    opts: T.RendererOpts;
+    api: T.RendererModelContainerApi;
+    savedHeight: number;
+    lineState: T.LineStateType;
+    updateHeight_debounced: (newHeight: number) => void;
+    ptyDataSource: (termContext: T.TermContextUnion) => Promise<T.PtyDataType>;
+    rawCmd: T.WebCmd;
+    loading: OV<boolean>;
+    isDone: OV<boolean>;
+    dirList: mobx.IObservableArray<any>;
+    version: OV<number>;
+    packetData: PacketDataBuffer;
+    curDirectory: string;
+    outputPos: number;
+
+    constructor() {
+        this.updateHeight_debounced = debounce(1000, this.updateHeight.bind(this));
+    }
+
+    initialize(params: T.RendererModelInitializeParams): void {
+        this.loading = mobx.observable.box(true, { name: "renderer-loading" });
+        this.isDone = mobx.observable.box(params.isDone, { name: "renderer-isDone" });
+        this.context = params.context;
+        this.opts = params.opts;
+        this.api = params.api;
+        this.savedHeight = params.savedHeight;
+        this.ptyDataSource = params.ptyDataSource;
+        this.rawCmd = params.rawCmd;
+        this.dirList = mobx.observable.array(null, {
+            name: "FileView-directorylist",
+        });
+        this.packetData = new PacketDataBuffer(this.packetCallback);
+        this.lineState = params.lineState;
+        this.curDirectory = this.lineState["prompt:file"];
+        this.outputPos = 0;
+    }
+
+    @boundMethod
+    packetCallback(packetAny: any) {
+        let packet: T.FileInfoType = packetAny;
+        if (packet == null) {
+            return;
+        }
+        this.curDirectory = packet.path;
+        this.outputPos = packet.outputpos;
```

Comment:
currently we store outputpos when we send commands, so that the new command can pick up where the old one left off. 

This honestly sucks and is gonna bring about a lot of timing issues as well as a lot of extra code, but I wasn't sure any other way to do this
Is there a way to get the current output pos from a file? 
I assume there are ways to get it with seek
I tried getting the file offset from the cirfile stat but that didn't work 

need to make a note to dig deeper into this and try and find a low level solution that doesn't involve sending output pos across the wire